homeassistant/components/subaru/entity.py

```diff
@@ -0,0 +1,60 @@
+"""Base class for all Subaru Entities."""
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify
+
+from .const import DOMAIN, ICONS, VEHICLE_NAME, VEHICLE_VIN
+
+
+class SubaruEntity(Entity):
+    """Representation of a Subaru Entity."""
+
+    def __init__(self, vehicle_info, coordinator):
+        """Initialize the Subaru Entity."""
+        self.coordinator = coordinator
+        self.car_name = vehicle_info[VEHICLE_NAME]
+        self.vin = vehicle_info[VEHICLE_VIN]
+        self.title = "entity"
+
+    @property
+    def name(self):
+        """Return name."""
+        return f"{self.car_name} {self.title}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return slugify(f"{DOMAIN} {self.vin} {self.title}")
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return ICONS.get(self.title)
+
+    @property
+    def should_poll(self):
```

Comment:
You can inherit from `CoordinatorEntity` which would take care of the boilerplate for you.

Reply:
Thanks! Change made.